src/benchmark.cc

```diff
@@ -193,22 +203,36 @@ void RunInThread(const benchmark::internal::Benchmark::Instance* b,
   manager->NotifyThreadComplete();
 }
 
-std::vector<BenchmarkReporter::Run> RunBenchmark(
+struct RunResults {
+  std::vector<BenchmarkReporter::Run> non_aggregates;
+  std::vector<BenchmarkReporter::Run> aggregates_only;
+
+  bool display_report_aggregates_only;
+  bool file_report_aggregates_only;
+};
+
+RunResults RunBenchmark(
     const benchmark::internal::Benchmark::Instance& b,
     std::vector<BenchmarkReporter::Run>* complexity_reports) {
-  std::vector<BenchmarkReporter::Run> reports;  // return value
+  RunResults run_results;
 
   const bool has_explicit_iteration_count = b.iterations != 0;
   size_t iters = has_explicit_iteration_count ? b.iterations : 1;
   std::unique_ptr<internal::ThreadManager> manager;
   std::vector<std::thread> pool(b.threads - 1);
   const int repeats =
       b.repetitions != 0 ? b.repetitions : FLAGS_benchmark_repetitions;
-  const bool report_aggregates_only =
-      repeats != 1 &&
-      (b.aggregation_report_mode == internal::ARM_Unspecified
-           ? FLAGS_benchmark_report_aggregates_only
-           : b.aggregation_report_mode == internal::ARM_ReportAggregatesOnly);
+  run_results.display_report_aggregates_only =
```

Comment:
why do you need to return this? you should be able to (as it was) have a local constant and decide to add to the vector or not based on the value.

Reply:
Then i don't understand the previous suggestion, https://github.com/google/benchmark/pull/665#discussion_r214663892.

Reply:
But to attempt to answer, `display_report_aggregates_only` is needed,
because i need to somehow convey back which result (all the reports, or aggregates only)
to report, for both of the reporters. I need to do so because i actually look at the
`benchmark::internal::Benchmark::Instance::aggregation_report_mode`,
which is per-instance, not just `FLAGS_benchmark_report_aggregates_only` /
`FLAGS_benchmark_display_aggregates_only`, which is global.

Reply:
Ah right, sorry.